File: yospace/src/main/java/com/bitmovin/player/integration/yospace/config/YospaceConfiguration.java
package com.bitmovin.player.integration.yospace.config;

import com.bitmovin.player.integration.yospace.YospaceLiveInitialisationType;

public class YospaceConfiguration {
    private String userAgent;
    private int readTimeout;
    private int connectTimeout;
    private int requestTimeout;
    private YospaceLiveInitialisationType liveInitialisationType;
    private boolean debug;

    public YospaceConfiguration() {
    }

    public YospaceConfiguration(String userAgent, int readTimeout, int connectTimeout, int requestTimeout, YospaceLiveInitialisationType liveInitialisationType, boolean debug) {
        this.userAgent = userAgent;
        this.readTimeout = readTimeout;
        this.connectTimeout = connectTimeout;
        this.requestTimeout = requestTimeout;
        this.liveInitialisationType = liveInitialisationType;
        this.debug = debug;
    }

    public String getUserAgent() {
        return userAgent;
    }

    public int getReadTimeout() {
        return readTimeout;
    }

    public int getConnectTimeout() {
        return connectTimeout;
    }

    public int getRequestTimeout() {
        return requestTimeout;
    }

    public YospaceLiveInitialisationType getLiveInitialisationType() {
        return liveInitialisationType;
    }

    public boolean isDebug() {
        return debug;
    }

}

File: yospace/src/main/java/com/bitmovin/player/integration/yospace/config/YospaceConfiguration.kt
package com.bitmovin.player.integration.yospace.config

import com.bitmovin.player.integration.yospace.YospaceLiveInitialisationType

data class YospaceConfiguration(
    val userAgent: String? = null,
    val readTimeout: Int = 25_000,
    val connectTimeout: Int = 25_000,
    val requestTimeout: Int = 25_000,
    val liveInitialisationType: YospaceLiveInitialisationType = YospaceLiveInitialisationType.DIRECT,
    val isDebug: Boolean = false,
    val filterMetadataType: MetadataType? = MetadataType.EMSG
)

public enum class MetadataType{
    ID3, EMSG
}